### ISSUE TITLE
Cannot create receivers in Explorer dashboard

### ISSUE THREAD

**Issue by vincentstchu** [rancher/dashboard#2095]
**What kind of request is this (question/bug/enhancement/feature request):**
question

**Steps to reproduce (least amount of steps as possible):**
Cannot add a recevier in the cluster explorer dashboard.
1. Acesss the cluster explorer -> Monitoring
2. Choose Receivers
3. Create Recevier. Choose recevier type: webhook and fill the blanks
4. Click create

**Result:**
Failed to create the receiver. It warns "e is undefined ".

![image](https://user-images.githubusercontent.com/24803055/101715228-04085780-3ad6-11eb-8b18-eff1f8e3cb85.png)

**Other details that may be helpful:**
We upgraded Rancher from v2.4.8 to v2.5.x. Current version is v2.5.3. Formerly in v2.4.8 the cluster level and project level monitoring are enabled in "Cluster Manger". After updating, we clean the  monitoring in "Cluster Manager" and switch to the new monitoring in "Cluster Explorer".

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.5.3
- Installation option (single install/HA): single

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Hosted
- Machine type (cloud/VM/metal) and specifications (CPU/memory): cloud (EC2)
- Kubernetes version (use `kubectl version`): 

```
Client Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-14T14:49:35Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.4", GitCommit:"d360454c9bcd1634cf4cc52d1867af5491dc9c5f", GitTreeState:"clean", BuildDate:"2020-11-11T13:09:17Z", GoVersion:"go1.15.2", Compiler:"gc", Platform:"linux/amd64"}
```

gz#13455

**Comment by tfiduccia** [comment]
Test Steps:
1.  Created rancher 2.4.8 on single docker machine
2. Create a linode cluster (3 nodes)
3. Enabled monitoring on both the cluster & in default project
4. Upgraded to 2.5.3
5. Went to dashboard

Issues:
- [ ] When I went to Apps & marketplace on linode cluster I didn't see any repositories, so I manually added rancher repo

Couldn't install monitoring when it's on in older version, so I turned it off for both cluster & default project.

- [ ] Noticed monitoring app still showing up in drop-down for the linode cluster.  I clicked on it and tried to create a receiver  and got an error 
![image](https://user-images.githubusercontent.com/11514927/102421946-e4d06500-3fc2-11eb-9202-26c162fdd40d.png)

- [ ] Went back to apps  & Marketplace and reinstalled monitoring & created Receiver.

There seems to be a few ways to really get this messed up, I mean his issue is probably a bug with the steps he took in upgrading and when he turned off monitoring in old rancher (if he ever did?), and if he reinstalled monitoring from charts in dashboard.  But the main thing is, I really don't think we should be showing the monitoring application in the drop-down if it hasn't been installed from dashboard.  It's very confusing and doesn't work properly.

**Comment by vincentstchu** [comment]
Just now I uninstalled Monitoring V2, deleted the namespace cattle-monitoring-system and then reinstalled again. It seems that the receivers could be added in dashboard now. Not quite sure about the reason since I did the same operations. Maybe there is a slight chance that this issue occurs? or Maybe there are some mistakes in my previous operations.

Anyway, it's glad to see it looks fine now. Many thanks that you guys take bothers to answer my questions. Appreciate your helps!

**Comment by vincentstchu** [comment]
@deniseschannon Should I close this issue or keep this issue open because it seems that there is still the chance of the issue comming back through previous comment?

**Comment by atsai1220** [comment]
> @deniseschannon Should I close this issue or keep this issue open because it seems that there is still the chance of the issue comming back through previous comment?

I still can't reliably create Receivers after upgrading monitoring.

**Comment by tfiduccia** [comment]
@atsai1220 - Is it that the receiver disappears, or it won't create and you get an error?  If it's an error can you please send a screenshot of that?

**Comment by atsai1220** [comment]
> @atsai1220 - Is it that the receiver disappears, or it won't create and you get an error? If it's an error can you please send a screenshot of that?

The receiver technically does not disappear from the alertmanager Secret but the UI does not show any Receivers and creating a new one shows `Cannot read property 'data' of undefined`.


![image](https://user-images.githubusercontent.com/14171643/102553160-418d5780-4077-11eb-80bc-8f67389565ca.png)

Contents of alertmanager secret
```yaml
global:
  resolve_timeout: 5m
receivers:
  - name: 'null'
  - slack_configs:
      - http_config: {}
        send_resolved: true
        api_url: >-
          https://hooks.slack.com/services/1234234/234234/234234
    email_configs: []
    webhook_configs: []
    name: test
route:
  group_by:
    - job
  group_interval: 5m
  group_wait: 30s
  receiver: 'null'
  repeat_interval: 12h
  routes:
    - match:
        alertname: Watchdog
      receiver: 'null'
templates:
  - /etc/alertmanager/config/*.tmpl

```


A complete re-install mention here (https://github.com/rancher/dashboard/issues/2095#issuecomment-743291851) allows Receivers to be created but this issue returns if I were to `upgrade` the release. Resulting in a need to completely re-install and lose all receivers and routes.

**Comment by vincentstchu** [comment]
Oh I see! It is upgrading that may cause the issue.  Today I upgraded the chart "rancher-monitoring" (just change some options or whatever), this issue showed up. 

P.S. I use firefox as browser so it warns `e is undefined`. In chrome it is `Cannot read property 'data' of undefined`

**Comment by vincentstchu** [comment]
Maybe duplication of #1845 . This will be fixed in v2.5.4

Upgraded Rancher from v2.5.3 to v2.5.4-rc6 and tested with followings:

1. Updated rancher-monitoring options
2. Added new receivers in dashboard
3. Upgraded rancher-monitoring chart from 9.4.201 to 9.4.202

The issue did not happen again.
